src/components/Collapsible/tests/Collapsible.test.tsx

```diff
@@ -1,88 +1,95 @@
 import React, {useState, useCallback} from 'react';
-// eslint-disable-next-line no-restricted-imports
-import {mountWithAppProvider} from 'test-utilities/legacy';
 import {mountWithApp} from 'test-utilities';
 import {Tokens} from 'utilities/theme';
 
 import {Collapsible, CollapsibleProps} from '../Collapsible';
 
 describe('<Collapsible />', () => {
-  const ariaExpandedSelector = '[aria-expanded=false]';
-
   it('does not render its children and indicates hidden with aria-hidden', () => {
-    const collapsible = mountWithAppProvider(
+    const collapsible = mountWithApp(
       <Collapsible id="test-collapsible" open={false}>
         content
       </Collapsible>,
     );
 
-    const hidden = collapsible.find(ariaExpandedSelector);
-    expect(hidden.exists()).toBe(true);
+    expect(collapsible).toContainReactComponent('div', {
+      'aria-expanded': false,
+    });
   });
 
   it('renders its children and does not render aria-hidden when open', () => {
-    const collapsible = mountWithAppProvider(
+    const collapsible = mountWithApp(
       <Collapsible id="test-collapsible" open>
         content
       </Collapsible>,
     );
 
-    const hidden = collapsible.find(ariaExpandedSelector);
-    expect(hidden.exists()).toBe(false);
-    expect(collapsible.contains('content')).toBe(true);
+    expect(collapsible).not.toContainReactComponent('div', {
+      'aria-expanded': false,
+    });
+
+    expect(collapsible).toContainReactComponent('div', {
+      children: expect.stringContaining('content'),
+    });
```

Comment:
I would just do this but what you have is fine

```suggestion
    expect(collapsible).toContainReactText('content');
```